lib/phoenix/router/scope.ex

```diff
@@ -33,6 +34,10 @@ defmodule Phoenix.Router.Scope do
       raise ArgumentError, "routes expect a module plug as second argument, got: #{inspect(plug)}"
     end
 
+    # Mark that we have routes defined in this scope
+    top = get_top(module)
+    put_top(module, %{top | has_routes?: true})
+
```

Comment:
I assume we can skip the second get_top call in line 41?

```suggestion
    # Get the top scope and mark that we have routes defined
    top = %{get_top(module) | has_routes?: true}
    put_top(module, top)
```